tests/tests_geomstats/test_pullback_metric.py

```diff
@@ -0,0 +1,307 @@
+"""Unit tests for the pull-back metrics."""
+
+import warnings
+
+import geomstats.backend as gs
+import geomstats.tests
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.geometry.pullback_metric import PullbackMetric
+
+
+class TestPullbackMetric(geomstats.tests.TestCase):
+    def setUp(self):
+        warnings.simplefilter('ignore', category=UserWarning)
+        gs.random.seed(0)
+        self.dim = 2
+        self.sphere = Hypersphere(dim=self.dim)
+        self.sphere_metric = self.sphere.metric
+
+        def _sphere_immersion(spherical_coords):
+            theta = spherical_coords[..., 0]
+            phi = spherical_coords[..., 1]
+            return gs.array([
+                gs.cos(phi) * gs.sin(theta),
+                gs.sin(phi) * gs.sin(theta),
+                gs.cos(theta)])
+
+        self.immersion = _sphere_immersion
+        self.pullback_metric = PullbackMetric(
+            dim=self.dim,
+            embedding_dim=self.dim + 1,
+            immersion=self.immersion
+        )
+
+    def test_immersion(self):
+        expected = gs.array([0., 0., 1.])
+        result = self.immersion(gs.array([0., 0.]))
+        self.assertAllClose(result, expected)
+
+        expected = gs.array([0., 0., -1.])
+        result = self.immersion(gs.array([gs.pi, 0.]))
+        self.assertAllClose(result, expected)
+
+        expected = gs.array([-1., 0., 0.])
+        result = self.immersion(gs.array([gs.pi / 2., gs.pi]))
+        self.assertAllClose(result, expected)
+
+    def test_immersion_and_spherical_to_extrinsic(self):
+        point = gs.array([0., 0.])
+        expected = self.immersion(point)
+        result = self.sphere.spherical_to_extrinsic(point)
+        self.assertAllClose(result, expected)
+
+        point = gs.array([0.2, 0.1])
+        expected = self.immersion(point)
+        result = self.sphere.spherical_to_extrinsic(point)
+        self.assertAllClose(result, expected)
+
+    @geomstats.tests.np_only
+    def test_jacobian_immersion(self):
+        def _expected_jacobian_immersion(point):
+            theta = point[..., 0]
+            phi = point[..., 1]
+            jacobian = gs.array([
+                [gs.cos(phi) * gs.cos(theta), - gs.sin(phi) * gs.sin(theta)],
+                [gs.sin(phi) * gs.cos(theta), gs.cos(phi) * gs.sin(theta)],
+                [-gs.sin(theta), 0.]
+            ])
+            return jacobian
```

Comment:
Why didn't you use `Hypersphere.tangent_spherical_to_extrinsic`?

Reply:
Same.